Source/ORTS.Menu/Consists.cs

```diff
@@ -56,116 +61,98 @@ internal Consist(string filePath, Folder folder)
             FilePath = filePath;
         }
 
-        internal Consist(string filePath, Folder folder, bool reverseConsist)
+        private static ITrainFile LoadConsist(string filePath)
         {
-            if (File.Exists(filePath))
+            switch (System.IO.Path.GetExtension(filePath).ToLowerInvariant())
             {
-                try
-                {
-                    var conFile = new ConsistFile(filePath);
-                    Name = conFile.Name.Trim();
-                    Locomotive = reverseConsist ? GetLocomotiveReverse(conFile, folder) : GetLocomotive(conFile, folder);
-                }
-                catch
-                {
-                    Name = "<" + catalog.GetString("load error:") + " " + System.IO.Path.GetFileNameWithoutExtension(filePath) + ">";
-                }
-                if (Locomotive == null) throw new InvalidDataException("Consist '" + filePath + "' is excluded.");
-                if (string.IsNullOrEmpty(Name)) Name = "<" + catalog.GetString("unnamed:") + " " + System.IO.Path.GetFileNameWithoutExtension(filePath) + ">";
+                case ".train-or":
+                    return TrainFile.LoadFrom(filePath);
+                case ".con":
+                    return new ConsistFile(filePath);
+                default:
+                    throw new InvalidDataException("Unknown train format");
             }
-            else
-            {
-                Name = "<" + catalog.GetString("missing:") + " " + System.IO.Path.GetFileNameWithoutExtension(filePath) + ">";
-            }
-            FilePath = filePath;
         }
 
         public override string ToString()
         {
             return Name;
         }
 
-        public static List<Consist> GetConsists(Folder folder)
+        public static List<Consist> GetConsists(Folder folder, IList<Folder> allFolders)
```

Comment:
I really am feeling like the work necessary for the "profile" property is excessive, for something that will have to be 100% replaced in the near future.

Reply:
If we're certain that profiles are absolutely not the way to go, I would be open to dropping this feature and reverting these changes.